hammett/utils/autodiscovery.py

```diff
@@ -6,26 +6,42 @@
 import inspect
 import pkgutil
 from types import ModuleType
+from typing import TYPE_CHECKING
 
 from hammett.core.screen import Screen
 
+if TYPE_CHECKING:
+    from collections.abc import Iterable
 
-def _autodiscover_screens_in_module(module: ModuleType) -> set[type[Screen]]:
+
+def _autodiscover_screens_in_module(
+    module: ModuleType,
+    exclude_screens: 'Iterable[type[Screen]]',
+) -> set[type[Screen]]:
     """Looks through the specified module for subclasses of the Screen class.
     The function skips Screen itself.
     """
 
     return {
         obj for _, obj in inspect.getmembers(module)
-        if inspect.isclass(obj) and issubclass(obj, Screen) and obj is not Screen
+        if inspect.isclass(obj)
+        and issubclass(obj, Screen)
+        and obj is not Screen
+        and obj not in exclude_screens
     }
 
 
-def autodiscover_screens(package_name: str) -> set[type[Screen]]:
+def autodiscover_screens(
+    package_name: str,
+    exclude_screens: 'Iterable[type[Screen]] | None' = None,
+) -> set[type[Screen]]:
```

Comment:
Wrap in single quotes 
```suggestion
) -> 'set[type[Screen]]':
```